src/search_algo.hpp

```diff
@@ -400,47 +406,100 @@ seedLooksPromising(LocalDataHolder<TGlobalHolder> const & lH,
     return false;
 }
 
-template <typename TGlobalHolder, typename TSeed>
+template <DbIndexType   c_indexType,
+          typename      TGlobalHolder,
+          typename      TSeed>
 inline void
 search_impl(LocalDataHolder<TGlobalHolder> & lH, TSeed && seed)
 {
-    namespace sc = seqan3::search_cfg;
-
-    seqan3::configuration cfg = sc::hit_all{}
-                              | sc::max_error_total{sc::error_count{static_cast<uint8_t>(lH.options.maxSeedDist)}}
-                              | sc::max_error_substitution{sc::error_count{static_cast<uint8_t>(lH.options.maxSeedDist)}}
-                              | sc::max_error_insertion{sc::error_count{0}}
-                              | sc::max_error_deletion{sc::error_count{0}}
-                              | sc::output_index_cursor{}
-                              | sc::on_result{[&lH] (auto && result)
-                                {
-                                    lH.cursor_buffer.push_back(result.index_cursor());
-                                }};
-
-    seqan3::search(seed, lH.gH.indexFile.index, cfg);
+
+    assert(lH.queries.size() == 1);
+    assert(lH.queries[0].size() == seed.size());
+
+    if constexpr (c_indexType == DbIndexType::FM_INDEX)
+    {
+
+        // prepare query for fmindex_collection, by shifting by one and reversing the string
+        for (size_t i{0}; i < seed.size(); ++i)
+        {
+            lH.queries[0][seed.size()-i-1] = seed[i].to_rank()+1;
+        }
```

Comment:
Do we need to copy all the seeds into rank-versions? Why does a view not work?

Reply:
You are always assiging to `lH.queries[0]`, so why is there a vector-of-vector at all?